### PR TITLE
fix(Image): images now use the MonochromeColor property on Android

### LINKED ISSUE #3533: [Android] BitmapIcon foreground property does not change the icon's color
## Current behavior
Bitmap icon remains black

## Expected behavior
Bitmap icon should be the color of its set foreground property

## How to reproduce it (as minimally and precisely as possible)
Open uno sample app in android navigate to BitmapIcon_Generic.xaml and notice all icon are black and the button on the page has no effect.

## Environment

Nuget Package:

Package Version(s):
Latest master
Affected platform(s):

- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [x] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by LHuss**
GitHub Issue (If applicable): fixes #3533

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Images on android do not use the MonochromeColor property, and thus by extension BitmapIcons do not get recolored when changing the Foreground property.

## What is the new behavior?

Images are recolored as needed.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs [hunk 1]
```diff
@@ -309,6 +311,12 @@ private async Task SetSourceResource(ImageSource newImageSource)
 			Func<CancellationToken, Task> setResource = async (ct) =>
 			{
 				_native.SetImageResource(newImageSource.ResourceId.Value);
+
+				if (MonochromeColor != null)
```
Comment (LHuss) [inline src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs:315]: There might be a better way/place to do this that I wasn't able to find, but (for now?) from what I gather, _native's base.SetImageResource handles all of the actual displaying given the source, including bitmap generation, so I had to wait for it to be called to recolor the bitmap and send it back. Insight would be appreciated

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs:315]: It's not terrible to have it here, but couldn't you do it in `NativeImage.SetImageResource()`?

File: src/Uno.UI/UI/Xaml/Controls/Image/NativeImage.Android.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ namespace Windows.UI.Xaml.Controls
 	internal partial class NativeImage : ImageView
 	{
 		private bool _skipLayoutRequest;
+		private bool _skipRecolor;
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Image/NativeImage.Android.cs:14]:
```suggestion
		private bool _skipRecolor;
		private bool NeedsRecolor => !_skipRecolor && Owner.MonochromeColor != null;
```

File: src/Uno.UI/UI/Xaml/Controls/Image/NativeImage.Android.cs [hunk 3]
```diff
@@ -67,6 +72,11 @@ public override void SetImageBitmap(Bitmap bm)
 			{
 				_skipLayoutRequest = true;
 
+				if (!_skipRecolor && Owner.MonochromeColor != null)
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Image/NativeImage.Android.cs:75]:
```suggestion
				if (NeedsRecolor)
```

File: src/Uno.UI/UI/Xaml/Controls/Image/NativeImage.Android.cs [hunk 4]
```diff
@@ -106,5 +116,53 @@ protected override bool SetFrame(int l, int t, int r, int b)
 
 			return base.SetFrame(l, t, r, b);
 		}
+
+		private void TryRecolorMonochrome()
+		{
+			if (!_skipRecolor && Owner.MonochromeColor != null && Drawable != null && Drawable is BitmapDrawable)
+			{
+				Bitmap source = ((BitmapDrawable)Drawable).Bitmap;
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Image/NativeImage.Android.cs:124]:
```suggestion
			if (NeedsRecolor && Drawable is BitmapDrawable bitmapDrawable)
			{
				var source = bitmapDrawable.Bitmap;
```

**Pull request merged**